aiida_cp2k/calculations/__init__.py

```diff
@@ -167,6 +168,21 @@ def prepare_for_submission(self, folder):
                                self.inputs.structure if 'structure' in self.inputs else None)
             write_basissets(inp, self.inputs.basissets, folder)
 
+        # if we have both basissets and structure we can start helping the user :)
+        if 'basissets' in self.inputs and 'structure' in self.inputs:
+            try:
+                scf_section = inp.get_section_dict('FORCE_EVAL/DFT/SCF')
+
+                if 'SMEAR' in scf_section and 'ADDED_MOS' not in scf_section:
+                    # now is our time to shine!
+                    added_mos = estimate_added_mos(self.inputs.basissets, self.inputs.structure)
+                    inp.add_keyword('FORCE_EVAL/DFT/SCF/ADDED_MOS', added_mos)
```

Comment:
Another thing that just popped up on my mind: I think we should put `override` to `False`. The users might want to provide the values through the input dictionary.

Reply:
That should be covered by the explicit check of finding `ADDED_MOS` in the `scf_section`, no?

Reply:
right.